testing/integration/fqdn_test.go

```diff
@@ -0,0 +1,248 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestFQDN(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		OS: []define.OS{
+			{Type: define.Linux},
```

Comment:
Restricting this to Linux for now because of the OS-specific ways of setting/getting a host's hostname.  The intention is to write the same test for all OS's, which will do in follow up PRs and refactor at that time to extract as much common code as possible.